### PR TITLE
main and feature branch export

### PR DESCRIPTION
**Pull request by td928**
to test the exporting fully on DO. Modifying the workflow to check if the branch Actions is being run on is main or feature branch.

### CODE REVIEW

**Comment by SashaWeinstein** [comment]
Awesome! Going to test this echoing the spaces variable to the console to check that it's correctly assigned. My only other comment is whether we want to echo all variables passed to the export function in that way

**Comment by SashaWeinstein** [comment]
Another comment here is that [facDB](https://github.com/NYCPlanning/db-facilities/blob/13447e36ce214bbc907c56b6f9d4740ef551e232/facdb.sh#L16) and [pluto](https://github.com/NYCPlanning/db-pluto/blob/0f078657098c6172c28620fc5c8e3eb43d3bfc8d/pluto_build/bin/config.sh#L16) get the branch name in bash instead of the github workflow. Might be nicer to use that approach instead

**Comment by SashaWeinstein** [comment]
Pushed a commit to fix syntax. I also don't remember why we have export census does but I don't know if we need it? Maybe it could come out in this PR

**Comment by SashaWeinstein** [comment]
Maybe export category and census can be two jobs in one build new combined build action?

**Comment by td928** [comment]
> Another comment here is that [facDB](https://github.com/NYCPlanning/db-facilities/blob/13447e36ce214bbc907c56b6f9d4740ef551e232/facdb.sh#L16) and [pluto](https://github.com/NYCPlanning/db-pluto/blob/0f078657098c6172c28620fc5c8e3eb43d3bfc8d/pluto_build/bin/config.sh#L16) get the branch name in bash instead of the github workflow. Might be nicer to use that approach instead

ah this is what I was looking for. Let me know if the most recent commit makes sense.

File: external_review/export_DO.sh [hunk 1]
```diff
@@ -5,8 +5,13 @@ function export_DO {
     echo $2
     geography_level=$1
     category=$2
-    filename="${category}_${geography_level}.csv"
-    SPACES="spaces/edm-publishing/db-eddt/${category}"
+    local filename="${category}_${geography_level}.csv"
+    local branchname=$(git rev-parse --symbolic-full-name --abbrev-ref HEAD)
```
Comment (SashaWeinstein) [inline external_review/export_DO.sh:9]: This looks good to me! I personally want to do more to improve the current export workflow but maybe this isn't the right time for that

**Review by SashaWeinstein — APPROVED** [review]
After further conversation with Te we decide to merge this in and wait to implement the bigger upgrade which is scoped out at #298

File: external_review/export_DO.sh [hunk 2]
```diff
@@ -5,8 +5,13 @@ function export_DO {
...
+    if [ "$branchname" = "main" ] ; then
+        SPACES="spaces/edm-publishing/db-eddt/${category}"
+    else
+        SPACES="spaces/edm-publishing/db-eddt/${branchname}/${category}"
+    fi
     mc cp .staging/$category/$filename $SPACES/$filename
     mc cp .staging/$category/metadata.yml $SPACES/metadata.yml
```
Comment (SashaWeinstein) [inline external_review/export_DO.sh:16]: Can we talk again about why not to just export all of `.staging/$category`? I think that would be easier to manage

Comment (td928) [reply external_review/export_DO.sh:16]: yeah I think my rationale for it is this eliminate the chances something unexpected ended up DO. It wasn't my design but I like it.

Comment (SashaWeinstein) [reply external_review/export_DO.sh:16]: Ok word. I think something unexpected ending up in DO is better than something we output not ending up there. This is why all the other pipelines are written to export everything I think. But it's ok either way I think we should run the action

**Comment by SashaWeinstein** [comment]
Ok I also see that some actions were run from `293-not-latest-from-DO` such as this one https://github.com/NYCPlanning/db-equitable-development-tool/actions/runs/2927282419. I think we should also stop running on pushes to the .yml, that's not a good design. I think that can change can come in this PR

**Comment by td928** [comment]
made another change on the `metadata.py` I think I got to the issue with str vs. int. And also got rid of the run on push in yml. Let me know if looks good on your end to merge. Thanks! @SashaWeinstein

File: .github/workflows/export_category.yml [hunk 3]
```diff
@@ -1,9 +1,6 @@
 name: Export Category 
 
 on:
-  push:
-    paths:
-      - ".github/workflows/export_category.yml"
```
Comment (SashaWeinstein) [inline .github/workflows/export_category.yml:6]: Awesome

**Review by SashaWeinstein — APPROVED** [review]
Great, think we can merge this in. We have this empty folder in DO, we should delete it right https://cloud.digitalocean.com/spaces/edm-publishing?i=266877&path=db-eddt%2F%2F

**Pull request merged**